### ISSUE TITLE
MdDialog doesn't have initial focus if opened by keyboard from MdMenu

### ISSUE THREAD

**Issue by chuanqisun** [angular/components#5909]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When opening a new dialog, the first focusable element in the dialog should be focused

#### What is the current behavior?
The focus is lost. press Tab key will focus on the next focusable element in the entire document

#### What are the steps to reproduce?
1. use IE11 (11.0.15055.0) or Edge (40.15055.0.0) Microsoft EdgeHTML 15.15055
2. open the plunker below
3. use Tab key to focus on the "Menu" button
4. press Enter key to open the menu
5. use Up/Down arrow key to select "Open dialog"
6. press Enter key to open dialog

observe: the input box in the dialog is not focused

7. press Tab key.

observe: the "Menu" button in the background is focused, confirming that the focus is outside the dialog.

Plunker template: https://plnkr.co/edit/eIbwTAn7qasHUwP4ERE2?p=preview

#### What is the use-case or motivation for changing an existing behavior?
a11y, usability for keyboard user, support for IE11, and Edge

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
the repro is based on the plunkers from the samples of MdMenu and MdDialog on https://material.angular.io/components.
all packages are from unpkg.com

#### Is there anything else we should know?
I still see the expected behavior in IE and Edge when the dialog is opened by mouse click
I see the expected behavior in Chrome when the dialog is opened by keyboard or by mouse

**Comment by jelbourn** [comment]
I believe this issue was resolved in a later version of Angular.
